commit 73b1f3d280dab048b0b02c64fbe671b92f639a74
Author: Eike Hein <hein@kde.org>
Date:   Mon Nov 18 23:59:03 2013 +0100

    Flush constraints events after location change.
    
    This allows the Panel containment to update the FormFactor property
    in time for control returning to a desktop script which performs
    prop changes dependant on the correct FormFactor value.

diff --git a/plasma/desktop/shell/scripting/panel.cpp b/plasma/desktop/shell/scripting/panel.cpp
index 4d30d35..11c2ae5 100644
--- a/plasma/desktop/shell/scripting/panel.cpp
+++ b/plasma/desktop/shell/scripting/panel.cpp
@@ -99,6 +99,8 @@ void Panel::setLocation(const QString &locationString)
     }
 
     c->setLocation(loc);
+
+    c->flushPendingConstraintsEvents();
 }
 
 PanelView *Panel::panel() const
